mne/io/curry/curry.py

```diff
@@ -0,0 +1,281 @@
+        idx = np.array([i in event_ids for i in curry_events[:, -1]])
+        curry_events = curry_events[idx]
+
+    return curry_events
+
+
+def read_raw_curry(input_fname, preload=False):
+    """
+    Read raw data from Curry files.
+
+    Parameters
+    ----------
+    input_fname : str
+        Path to a curry file with extensions .dat, .dap, .rs3, .cdt, cdt.dpa,
+        .cdt.cef or .cef.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory). If the curry file
+        is stored in ASCII data format, then preload will automatically be
+        set to True.
+
+    Returns
+    -------
+    raw : instance of RawCurry
+        A Raw object containing Curry data.
+
+    """
+    # we don't use os.path.splitext to also handle extensions like .cdt.dpa
+    fname_base, ext = input_fname.split(".", maxsplit=1)
+
+    curry_vers = _get_curry_version(ext)
+    _check_missing_files(fname_base, curry_vers)
+
+    info, n_trials, n_samples, curry_vers, ascii = _read_curry_info(fname_base,
+                                                                    curry_vers)
+
+    raw = RawCurry(fname_base + DATA_FILE_EXTENSION[curry_vers], info,
+                   n_samples, ascii, preload)
+
+    return raw
+
+
+class RawCurry(BaseRaw):
+    """
+    Raw object from Curry file.
+
+    Parameters
+    ----------
+    data_fname : str
+        Path to the Curry data file (.dat or .cdt).
+    info : instance of mne.Info
+        An instance of mne.Info as returned by mne.io.curry._read_curry_info.
+    n_samples : int
+        The number of samples of the curry data file.
+    ascii : bool (default False)
+        If the loaded data file is coded in ASCII, ascii must be set to True.
+        Else must be False.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory) If ascii is True,
+        preload will automatically be set to True.
+    %(verbose)s
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+
+    """
+
+    def __init__(self, data_fname, info, n_samples, ascii,
+                 preload=False, verbose=None):
+
+        data_fname = os.path.abspath(data_fname)
+
+        last_samps = [n_samples - 1]
+
+        if ascii:
+            if not preload:
+                warn('Got ASCII format data as input. Data will be preloaded.')
```

Comment:
this seem unecessary as you can use skiprows and max_rows params in loadtxt:

https://docs.scipy.org/doc/numpy/reference/generated/numpy.loadtxt.html